ce/api/geo.py

```diff
@@ -67,14 +68,9 @@ def __init__(self, func, maxsize=50):
 
     def __call__(self, *args):
 
-        nc, wkt = args
-        # If we have no key, automatic cache miss
-        if (not hasattr(nc, 'model_id')):
-            log.debug('Cache MISS (attribute \'model_id\' not found)')
-            return self.func(*args)
+        fname, wkt, varname = args
 
-        # Set key to model_id and wkt polygon
-        key = (nc.model_id, wkt)
+        key = (fname, wkt)
```

Comment:
What's your motivation for this? We're going to have situations where it generates the same mask for the same grid multiple times again. We know that model_id isn't idea and we'd rather key on the grid_id as discussed, but this is a separate issue.